src/components/Autocomplete/OnSubmitContext.tsx

```diff
@@ -0,0 +1,34 @@
+import { createContext } from "preact"
+import { useContext, useMemo } from "preact/hooks"
+import { ComponentChildren } from "preact"
+import { SearchAnalyticsOptions } from "@nosto/nosto-js/client"
+
+type OnSubmitContextType = {
```

Comment:
make the onSubmit function directly the context type instead of a wrapped version